src/libraries/System.Collections.NonGeneric/tests/ArrayListTests.cs

```diff
@@ -84,17 +84,8 @@ public static void DebuggerAttribute()
             DebuggerAttributes.ValidateDebuggerDisplayReferences(new ArrayList());
             DebuggerAttributes.ValidateDebuggerTypeProxyProperties(new ArrayList() { "a", 1, "b", 2 });
 
-            bool threwNull = false;
-            try
-            {
-                DebuggerAttributes.ValidateDebuggerTypeProxyProperties(typeof(ArrayList), null);
-            }
-            catch (TargetInvocationException ex)
-            {
-                threwNull = ex.InnerException is ArgumentNullException;
-            }
-
-            Assert.True(threwNull);
+            TargetInvocationException tie = Assert.Throws<TargetInvocationException>(() => DebuggerAttributes.CreateDebuggerTypeProxyWithNullArgument(typeof(ArrayList)));
```

Comment:
Just trying to undertand how non-generic collections are related to this change. Is it part of an infrastructural refactoring? Is ReadOnlyDictionary used under wraps somehow?